algorithms/src/snark/varuna/tests.rs

```diff
@@ -524,3 +524,293 @@ mod varuna_hiding {
+    use std::{collections::BTreeMap, fs, ops::Deref, path::PathBuf, str::FromStr, sync::Arc};
+
+    type FS = crate::crypto_hash::PoseidonSponge<Fq, 2, 1>;
+    type MM = VarunaNonHidingMode;
+    type VarunaSonicInst = VarunaSNARK<Bls12_377, FS, MM>;
+
+    /// Returns the path to the `resources` folder for this module.
+    fn resources_path() -> PathBuf {
+        // Construct the path for the `resources` folder.
+        let mut path = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+        path.push("src");
+        path.push("snark");
+        path.push("varuna");
+        path.push("resources");
+
+        // Create the `resources` folder, if it does not exist.
+        if !path.exists() {
+            fs::create_dir_all(&path).unwrap_or_else(|_| panic!("Failed to create resources folder: {path:?}"));
+        }
+        // Output the path.
+        path
+    }
+
+    /// Loads the given `test_folder/test_file` and asserts the given `candidate` matches the expected values.
+    #[track_caller]
+    fn assert_snapshot<S1: Into<String>, S2: Into<String>>(test_folder: S1, test_file: S2, candidate: &str) {
+        // Construct the path for the test folder.
+        let mut path = resources_path();
+        path.push(test_folder.into());
+
+        // Create the test folder, if it does not exist.
+        if !path.exists() {
+            fs::create_dir(&path).unwrap_or_else(|_| panic!("Failed to create test folder: {path:?}"));
+        }
+
+        // Construct the path for the test file.
+        path.push(test_file.into());
+        path.set_extension("txt");
+
+        // Create the test file, if it does not exist.
+        if !path.exists() {
+            panic!("Test file does not exist: {path:?}");
+        }
+
+        println!("Testing path: {:?}", path);
+
+        // Assert the test file is equal to the expected value.
+        expect_test::expect_file![path].assert_eq(candidate);
+    }
+
+    // In the case that we're creating a test vectors from a trusted branch
+    fn create_test_vector(folder: &str, file: &str, data: &str) {
+        // Construct the path for the test folder.
+        let mut path = resources_path();
+        path.push(folder);
+
+        // Create the test folder, if it does not exist.
+        if !path.exists() {
+            fs::create_dir(&path).unwrap_or_else(|_| panic!("Failed to create test folder: {path:?}"));
+        }
+
+        // Construct the path for the test file.
+        path.push(file);
+        path.set_extension("txt");
+
+        // Create the test file, if it does not exist.
+        if !path.exists() {
+            fs::File::create(&path).unwrap_or_else(|_| panic!("Failed to create file: {path:?}"));
+        }
+
+        fs::write(&path, data).unwrap_or_else(|_| panic!("Failed to write to file: {:?}", path));
+    }
+
+    #[test]
+    fn test_proving_system_with_test_vectors() {
+        run_prover_sage_vectors(false);
+    }
+
+    #[test]
+    #[ignore]
```

Comment:
Good point - I will create a feature flag for this. There's limited risk I think ultimately because of the PR flow, but the feature flag will ensure we're making an intent choice to update the test vectors.